lib/net.c

```diff
@@ -1200,12 +1201,99 @@ start_iface(struct gatekeeper_if *iface, unsigned int num_attempts_link_get)
 	return ret;
 }
 
+static void
+calculate_net_config_para(struct net_config *net_conf)
+{
+	int i;
+
+	const char *filtered_blocks[] = { "dynamic_conf" };
+
+	/*
+	 * The total number of lcores used by any functional block
+	 * for either RX or TX except for Dynamic Configuration.
+	 */
+	int gatekeeper_num_lcores = launch_count_lcores(filtered_blocks,
+		RTE_DIM(filtered_blocks));
+
+	/*
+	 * The total number of receive descriptors to
+	 * allocate for the receive ring over all interfaces.
+	 */
+	uint16_t gatekeeper_total_rx_desc = net_conf->front.num_rx_queues *
+		net_conf->front.num_rx_desc + (net_conf->back_iface_enabled ?
+		net_conf->back.num_rx_queues * net_conf->back.num_rx_desc : 0);
+
+	/*
+	 * The total number of transmit descriptors to
+	 * allocate for the transmit ring over all interfaces.
+	 */
+	uint16_t gatekeeper_total_tx_desc = net_conf->front.num_tx_queues *
+		net_conf->front.num_tx_desc + (net_conf->back_iface_enabled ?
+		net_conf->back.num_tx_queues * net_conf->back.num_tx_desc : 0);
+
+	/*
+	 * The number of elements in the mbuf pool.
+	 *
+	 * The optimum size (in terms of memory usage) for a mempool is when
+	 * it is a power of two minus one.
```

Comment:
Adjust and move this comment to `net_conf->gatekeeper_num_mbuf` since it is being applied to `gatekeeper_max_num_pkt` right now.